jcsv.go

```diff
@@ -1,11 +1,107 @@
 package jcsv
 
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+func removeDoubleQuotes(str string) string {
+	str = strings.Replace(str, "\"", " ", -1)
+	return str
+}
+
 func JsonToCsv(j []byte, addHeaders bool) ([]byte, error) {
-	// TODO: convert JSON data in `j` into CSV format and return
-	return nil, nil
+	if j == nil {
+		return nil, nil
+	}
+	var header string
+	i := 0
+	begin := 0
+	if addHeaders {
+		for ; j[i] != '}'; i++ {
+			if j[i] == ':' {
+				header += strings.Trim(string(j[begin:i]), "\"  ") + ","
+				begin = i + 1
+			}
+			if j[i] == '\n' || j[i] == ',' {
+				begin = i + 1
+			}
+		}
+		header = strings.TrimRight(header, ",")
+	}
+	var objectFinish bool = false
+	var result string
+	result += header + "\n"
+	i = 0
+	begin = 0
+	for ; i < len(j); i++ {
+		if j[i] == ':' {
+			objectFinish = false
+			begin = i + 1
+		}
+		if j[i] == ',' && objectFinish == false {
+			result += strings.Trim(string(j[begin:i]), "\"\n\t ") + ","
+		}
+		if j[i] == '}' {
+			//result += strings.Trim(string(j[begin:i]), "\"\n\t ") + "\n"
+			result += strings.Trim(removeDoubleQuotes(string(j[begin:i])), "\"\n\t ") + "\n"
+			objectFinish = true
+		}
+	}
+
+	return []byte(result), nil
 }
 
 func CsvToJson(c []byte, hasHeaders bool) ([]byte, error) {
-	// TODO: convert CSV data in `c` into JSON format and return
-	return nil, nil
+	if c == nil {
+		return nil, nil
+	}
+	var attributes []string
+	begin := 0
+	keysCount := 1
```

Comment:
There are no comments to understand what these variables are for. Add some comments, please.

Reply:
Sure, in next push I will comment complete code.